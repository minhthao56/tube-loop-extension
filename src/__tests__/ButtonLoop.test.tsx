/* eslint-disable testing-library/no-node-access */
import { render } from "@testing-library/react";
import ButtonLoop from "../ButtonLoop";
import useEnableLoopVideo from "../hooks/useEnableLoopVideo";
import useMessageToggle from "../hooks/useMessageToggle";
import userEvent from "@testing-library/user-event";

jest.mock("../hooks/useEnableLoopVideo", () => jest.fn());
jest.mock("../hooks/useMessageToggle", () => jest.fn());

describe("<ButtonLoop/>", () => {
  const handleToggleLoopAttributeVideo = jest.fn();
  const renderComponent = () => {
    return render(<ButtonLoop />);
  };

  beforeEach(() => {
    (useMessageToggle as jest.Mock).mockReturnValue({
      isEnableAlwaysLoop: false,
      isEnableButtonLoop: true,
    });

    (useEnableLoopVideo as jest.Mock).mockReturnValue({
      handleToggleLoopAttributeVideo,
      isLooped: true,
    });
  });

  it("should match snapshot", () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
    expect(document.querySelector(".ytp-button")).toBeInTheDocument();
  });

  it("should be call handleToggleLoopAttributeVideo when clicking button loop", () => {
    renderComponent();
    const button = document.querySelector(".ytp-button") as HTMLElement;
    userEvent.click(button);
    expect(handleToggleLoopAttributeVideo).toBeCalledTimes(1);
  });

  it("should have style opacity is 0.6 when isLooped is false", () => {
    (useEnableLoopVideo as jest.Mock).mockReturnValue({
      handleToggleLoopAttributeVideo,
      isLooped: false,
    });
    renderComponent();
    const svg = document.querySelector(".ytp-button")?.querySelector("svg");
    expect(svg).toHaveStyle("opacity:0.6");
  });

  it("should hide the button loop when isEnableButtonLoop is false", () => {
    (useMessageToggle as jest.Mock).mockReturnValue({
      isEnableAlwaysLoop: false,
      isEnableButtonLoop: false,
    });
    renderComponent();
    expect(document.querySelector(".ytp-button")).not.toBeInTheDocument();
  });
});
