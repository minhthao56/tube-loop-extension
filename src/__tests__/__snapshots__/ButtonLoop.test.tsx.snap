// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonLoop/> should match snapshot 1`] = `
<div>
  <div
    class="ytp-button"
  >
    <div
      style="height: 100%; display: flex; justify-content: center; align-items: center;"
    >
      <svg
        fill="currentColor"
        height="26"
        stroke="currentColor"
        stroke-width="0"
        style="opacity: 1;"
        viewBox="0 0 24 24"
        width="26"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M21 6h-5v2h4v9H4V8h5v3l5-4-5-4v3H3a1 1 0 0 0-1 1v11a1 1 0 0 0 1 1h18a1 1 0 0 0 1-1V7a1 1 0 0 0-1-1z"
        />
      </svg>
    </div>
  </div>
</div>
`;
